src/python/WMCore/WMSpec/WMTask.py

```diff
@@ -331,6 +332,41 @@ def build(self, workingDir):
         master(self)
         return
 
+    def addEnvironmentVariables(self, envDict):
+        """
+        _addEnvironmentVariables_
+
+        add a key = value style setting to the environment for this task and all
+        its children
+        """
+        for key, value in viewitems(envDict):
+            setattr(self.data.environment, key, value)
+        for task in self.childTaskIterator():
+            task.addEnvironmentVariables(envDict)
+        return
+
+    def setOverrideCatalog(self, tfcFile):
+        """
+        _setOverrideCatalog_
+
+        Used for setting overrideCatalog option for each step in the task.
+        """
+        for step in self.steps().nodeIterator():
+            if step.stepType == "CMSSW":
```

Comment:
Due to this check, overrideCatalog is no longer being aplied to LogCollect steps. To make it work I can just add an "or" to include "WMStep" and that's it. But, probably, the best solution would involve to add setOverrideCatalog() to either `LogCollectStepHelper` or `CoreHelper` (as right now it only exists in CMSSWStepHelper). What do you think @amaltaro?

Reply:
German, I'm afraid I do not know what would be the best option here. Would you mind providing an initial commit with the best candidate you can think of? If it looks solid, then you can run further tests and proceed with that.

Reply:
@amaltaro I moved `setOverrideCatalog()` from CMSSWHelper to CoreHelper, as now it is needed in at least 2 different step types and I wanted to avoid code dups. I tested the current versión an it works as expected. So the PR is ready for you to review.